### PR TITLE
feature/26744/Fix-request-uncofirmed

### PR DESCRIPTION
**Pull request by massimez**

### CODE REVIEW

File: pages/index.vue [hunk 1]
```diff
@@ -2,10 +2,24 @@
...
+    }),
+  },
+  async created() {
+    const access = this.$cookies.get('access');
+    const userStatus = this.$cookies.get('userStatus');
+    if (+userStatus === UserStatuses.Confirmed && access) {
+      if (!this.userData.id) await this.$store.dispatch('user/getUserData');
+      if (this.userData.role === UserRole.EMPLOYER) await this.$router.push(Path.WORKERS);
+      else if (this.userData.role === UserRole.WORKER) await this.$router.push(Path.QUESTS);
+    } else await this.$router.push('/sign-in');
```
Comment (matykoks) [inline pages/index.vue:22]: Path

File: middleware/auth.js [hunk 2]
```diff
@@ -25,10 +25,10 @@ export default async function ({
       await store.dispatch('user/logout');
       return redirect(Path.SIGN_IN);
     }
-    if (userData.id === '') {
+    if (userData.id === '' && +userStatus === UserStatuses.Confirmed) {
```
Comment (matykoks) [inline middleware/auth.js:28]: we have getter isAuth

File: pages/index.vue [hunk 3]
```diff
@@ -2,10 +2,24 @@
...
+  computed: {
+    ...mapGetters({
+      userData: 'user/getUserData',
+    }),
+  },
+  async created() {
+    const access = this.$cookies.get('access');
+    const userStatus = this.$cookies.get('userStatus');
+    if (+userStatus === UserStatuses.Confirmed && access) {
+      if (!this.userData.id) await this.$store.dispatch('user/getUserData');
```
Comment (matykoks) [inline pages/index.vue:19]: here i think you can use getter isAuth

File: middleware/auth.js (hunk 2 above)
Comment (massimez) [reply middleware/auth.js]: Here we are checking if user data have been fetched, and isAuth is only about the access token

**Pull request merged**